CAIPs/caip-122.md

```diff
@@ -37,7 +37,7 @@ The data model _MUST_ contain the following fields:
 | Name              | Type            | Mandatory | Description                                                                                                                                                                                       |
 | ----------------- | --------------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `domain`          | string          | ✓         | [RFC 4501][rfc 4501] `dnsauthority` that is requesting the signing.                                                                                                                               |
-| `address`         | string          | ✓         | Blockchain address, as defined by [CAIP-10][caip-10], performing the signing; should include [CAIP-2][caip-2] chain id namespace                                                                  |
+| `account_id`         | string          | ✓         | Blockchain address performing the signing, expressed as the `account_id` segment of a [CAIP-10][caip-10]; should NOT include [CAIP-2][caip-2] `chain_id`.| namespace                                                                  |
```

Comment:
@bumblefudge  In CAIP-10, `account_id` is a whole identifier (this is what I was suggesting in my previous comment). The segment you mention here is called `account_address`: https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-10.md#syntax

Conversion of CAIP-10 identifier to address and chain ID is already described at lines 76, 87 and 97:

>`account_address(address)` is the `account_address` segment of a [CAIP-10][] `address` of the data model
>`chain_id(address)` is a `chain_id` part of [CAIP-10][caip-10] `address` of the data model.

Reply:
amazing how much i've confused myself with this.  it now matches the other section, apologies.

making `account` the whole [machine-readable] CAIP-10 doesn't work very well for clients (like all current SIWE clients) that display account to the end-user in native form (i.e. without the CAIP-2 prefix, which can be cryptic and opaque in many cases), thus the account_address only design choice to date.

Reply:
The text still doesn't make sense and contradicts itself.

Reply:
I agree with @silverpill I think I did it the right way in the tezos-caip-122